src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs

```diff
@@ -26,6 +26,7 @@ public async Task EvaluateAsync(CancellationToken cancellationToken)
             if (_restartImmediatelySessionPreference.HasValue)
             {
                 await GetRudeEditResult(_restartImmediatelySessionPreference.Value, cancellationToken);
+                return;
```

Comment:
If the user selected a Always option, the await would have terminated quickly, but then we immediately prompt them on Line 31.